### PR TITLE
[Feature]: Auto-configure autodoc reference links

### PR DESCRIPTION
**Pull request by CodyCBakerPhD**
@bendichter This code automatically forms reference links from the autodocs running on all the check modules.

The only thing that has to be specified manually is the hand-written Best Practices section that links to the API. E.g., 

```
.. _best_practice_function_name

Best Practice Section
~~~~~~~~~~~~~~~~
...

Check function :ref:`check_function_name <check_function_name>`
```

### CODE REVIEW

File: docs/api/nwbinspector.rst [hunk 1]
```diff
@@ -9,7 +9,12 @@ Core Library
     :maxdepth: 4
 
 .. autofunction:: nwbinspector.nwbinspector.inspect_all
+    :noindex:
```
Comment (CodyCBakerPhD) [inline docs/api/nwbinspector.rst:12]: For suppressing a certain error that occurs during sphinx build

**Comment by bendichter** [comment]
@CodyCBakerPhD could you make the docs auto-build here as well?

**Comment by CodyCBakerPhD** [comment]
@bendichter It's supposed to be and was previously working on this repo, too. Trying to figure out why it's not displaying in the CI tab

**Comment by CodyCBakerPhD** [comment]
OK it's showing up now. The trigger conditions are a bit touchy it seems, we'll see if it keeps being an issue

**Comment by bendichter** [comment]
@CodyCBakerPhD Would this create a link from the API docs to the best practices, or just from the best practices to the API docs?

**Comment by CodyCBakerPhD** [comment]
> @CodyCBakerPhD Would this create a link from the API docs to the best practices, or just from the best practices to the API docs?

@bendichter What is automatically formed is (i) a reference **for** a particular API function loaded via `automodule`. The alternative for that is to manually add the reference applied to an `autofunction` (one for each check), or otherwise manually adding it to each class docstring, and (ii) a link **from** the API function **to** its section in the Best Practices.

This is exemplified by the Time Series practice for regular time stamps: https://nwbinspector--101.org.readthedocs.build/en/101/api/checks.html#nwbinspector.checks.time_series.check_regular_timestamps

**Comment by bendichter** [comment]
@CodyCBakerPhD ok, cool. Let's go over the logic in our meeting today

File: docs/best_practices/tables.rst [hunk 2]
```diff
@@ -13,6 +14,11 @@ Store data with long columns rather than long rows. When constructing dynamic ta
 inefficient to store data in a table with many columns.
 bools
 
+Check function :ref:`check_column_binary_capability <check_column_binary_capability>`
+
```
Comment (bendichter) [inline docs/best_practices/tables.rst]:
```suggestion
:py:func:`nwbinspector.checks.table.check_column_binary_capability`
```

**Comment by CodyCBakerPhD** [comment]
#105 includes these changes but will continue to try to adapt the PyNWB approach

**Pull request closed without merging**